src/ios/GeolocationPlugin.m

```diff
@@ -0,0 +1,276 @@
+//
+//  GeolocationPlugin.m
+//  Geolocation
+//
+//  Created by dev on 2/9/16.
+//
+//
+
+#import "GeolocationPlugin.h"
+
+@implementation GeolocationPlugin
+
+-(void)start:(CDVInvokedUrlCommand *)command
+{
+    coord1 = [command.arguments objectAtIndex:0];   // Get center and radio of region1
+    coord2 = [command.arguments objectAtIndex:1];   // Get center and radio of region2
+    url = [command.arguments objectAtIndex:2];      // Get server url
+
+    
+    [self stopTimer];
+    [self startTimedTask];
+}
+
+-(void)startTimedTask{
+    time_interval = [NSTimer scheduledTimerWithTimeInterval:75.0 target:self selector:@selector(performBackgroundTask) userInfo:nil repeats:YES];
+}
+-(void)performBackgroundTask{
+    
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self initData];
+        });
+    });
+}
+
+-(void)stopTimer{
+    if(time_interval)
+    {
+        [time_interval invalidate];
+        time_interval = nil;
+        [self stopUpdatingLocationWithMessage:NSLocalizedString(@"Error", @"Error")];
+    }
+}
+
+-(void)stop:(CDVInvokedUrlCommand *)command
+{
+    [self stopTimer];
+}
+
+-(void)initData{
+    [self createDataBase];
+    
+    // Add location Manager class
+    if (nil == self.locationManager) {
+        self.locationManager = [[CLLocationManager alloc] init];
+        // Get version of iOS
+        if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 8.0) {
+            
+            [ self.locationManager requestWhenInUseAuthorization];
+            
+            [self.locationManager requestAlwaysAuthorization];
+        }
+    }
+    now     = [NSDate date];
+    now1    = [NSDate date];
```

Comment:
doesn't look like this is used anywhere else? if so please remove. anyhow, `now` is ok as a name, but `now1` is not because you can't tell what it means. If needed better make like `nowAtStart` or `startedAt` and `initNow`...